packages/messaging/lib/index.d.ts

```diff
@@ -162,6 +183,35 @@ export namespace FirebaseMessagingTypes {
     senderId?: string;
   }
 
+  /**
+   * Options for `getToken()`
+   */
+  export interface GetTokenOptions {
+    /**
+     * The VAPID key used to authenticate the push subscribers
+     *  to receive push messages only from sending servers
+     * that hold the corresponding private key.
+     *
+     * @platform web
+     */
+    vapidKey?: string;
+
+    /**
+     * The service worker registration for receiving push messaging.
+     * If the registration is not provided explicitly, you need to
+     * have a firebase-messaging-sw.js at your root location.
+     *
+     * @platform web
+     */
+    serviceWorkerRegistration?: ServiceWorker;
```

Comment:
I think `ServiceWorkerRegistration` is the upstream type here, not `ServiceWorker` 🤔  ?

https://github.com/firebase/firebase-js-sdk/blob/49ee786f2b022e65aef45693e1a8b546d889ec10/packages/messaging/src/interfaces/public-types.ts#L134

```suggestion
    serviceWorkerRegistration?: ServiceWorkerRegistration;
```